text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,203 @@
+
+Before buildpack execution the platform reads the `sha256` values from the label metadata on `new-buildpack`. It then searches through local images to check for:
+  1) Exposed `asset` sha256 values.
+  2) A match between the expose, and the `sha256` value from `new-buildpack`.
+
+If both conditions are met, the platform applies the `layerID` mapped to this `sha256` to the build environment, exposing a vendored asset at `/cnb/assets/<asset-sha256>` to all buildpacks.
+
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+### Asset Image Definition.
+
+An Asset Image is a collection of layers each of which provide one or more assets. These layers are in lexicographic order by `diffID`.
+
+```
+<layer1> ━━ /cnb/deps/<java11-asset-sha256>/java11
+<layer2> ━━ /cnb/deps/<java13-asset-sha256>/java13
+
+...
+<layern> ━━ /cnb/deps/<java15-asset-sha256>/java15
+```
+
+Each Asset Image will have a `io.buildpacks.buildpack.assets` Label. This label's contents will be a `json` object mapping each `asset` sha256 to a `JSON` object that contains a `layerDiffID`, and other optional keys.
+
+#### Example
+
+``` json
+{
+
+  "<java11-asset-sha256>" : {
+     "layerDiffID": "<layer1-diffID>",
+     "additional_paths": ["other/endpoint1", "other/endpoint2"],
+     "metadata": {}
+  },
+  "<java13-asset-sha256>" : {
+    "layerDiffID": "<layer2-diffID>",
+
+    "additional_paths": [],
+    "metadata": {
+      "name": "java13"
+    }
+  },
+  "<java15-asset-sha256>" : {
+    "layerDiffID": "<layerk-diffID>",
+    "additional_paths": []
+  }
+}
+
+```
+
+#### Schema
+TODO: once above format is validated.
+
+
+
+### Creating Asset Images.
+
+Asset Image creating will be performed by a new pack command, `pack packages-assets`. `package-assets` and will use an `assets.toml` configuration file. Each entry in `asset.toml` will provide the following:
+  - `uri` string (required)
+  - `sha256` (required) used for validation
+  - `additional_paths` array
+  - `metadata` arbitrary mapping
+
+The `additional_paths`array lists additional endpoints where the platform should provide the asset to buildpack during a buildpack run. This is provided for cases where `assets` require absolute paths.
+
+
+Ever assets must be in one of the following archive formats:
+  - `tar.gz`
+  - ...
+
+
+So the `asset.toml` to produce the asset image [above](#Example) would be:
+
+#### Example
+``` toml
+[[asset]]
+url = "https://path/to/java11.tgz"
+sha256 = "some-sha256"
+additional_paths = ["other/endpoint1", "other/endpoint2"]
+
+[[asset]]
+url = "/local/path/to/java13.tgz"
+sha256 = "another-sha256"
+  [metadata]
+    name = "java13"
+ 
+[[asset]]
+...
+
+[[asset]]
+url = "/local/path/to/java15.tgz"
+sha256 = "another-nother-sha256"
+```
+
+The `pack create-asset` command will
+  - download all assets, and validate a sha256 of the artifact.
```

Comment:
Is download reasonable?  What if you want to build an asset image with an asset you currently have, but cannot put on then HTTP endpoint?